src/v/http/probe.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include <boost/beast/http.hpp>
+
+#include <cstdint>
+
+namespace http {
+
+struct client_probe {
+    using verb = boost::beast::http::verb;
+
+    /// Total inbound traffic size
+    uint64_t in;
+    /// Total outbound traffic size
+    uint64_t out;
+    /// Number of active downloads
+    int64_t get_requests;
+    /// Number of active uploads
+    int64_t put_requests;
+    /// All active requests
+    int64_t all_requests;
+    /// Number of connection errors
+    uint64_t transport_errors;
+
+    /// Register incomming traffic
+    void add_inbound_bytes(uint64_t bytes) { in += bytes; }
+    /// Register outgoing traffic
+    void add_outbound_bytes(uint64_t bytes) { out += bytes; }
+    /// Register the begining of the request
+    void begin_request(verb v) {
```

Comment:
this isn't RAII safe. 

why not return a struct - see hdr_hist.h for example ctor increments, dtor decrements

Reply:
did exactly that